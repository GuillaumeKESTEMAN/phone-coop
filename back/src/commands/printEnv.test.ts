import initPrintEnvCommand from './printEnv';
import type { WhookCommandArgs } from '@whook/cli';

describe('printEnvCommand', () => {
  const log = jest.fn();

  beforeEach(() => {
    log.mockReset();
  });

  it('should work', async () => {
    // Have to use Object.assign for some reason here
    // See : https://stackoverflow.com/questions/56349619/ts2352-declare-object-with-dynamic-properties-and-one-property-with-specific-t
    const printEnvCommand = await initPrintEnvCommand({
      log,
      ENV: { NODE_ENV: 'test' },
      args: Object.assign(
        {
          keysOnly: true,
        },
        {
          _: ['printEnv'],
        },
      ) as unknown as WhookCommandArgs,
    });
    const result = await printEnvCommand();

    expect({
      result,
      logCalls: log.mock.calls.filter(([type]) => !type.endsWith('stack')),
    }).toMatchSnapshot();
  });
});
