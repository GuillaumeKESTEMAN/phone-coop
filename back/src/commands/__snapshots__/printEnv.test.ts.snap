// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printEnvCommand should work 1`] = `
Object {
  "logCalls": Array [
    Array [
      "info",
      "[\\"NODE_ENV\\"]",
    ],
  ],
  "result": undefined,
}
`;
